File: examples/SignatureRequestCreateEmbeddedGroupedSignersExample.php
<?php

namespace Dropbox\SignSandbox;

require_once __DIR__ . '/../vendor/autoload.php';

use SplFileObject;
use Dropbox;

$config = Dropbox\Sign\Configuration::getDefaultConfiguration();
$config->setUsername("YOUR_API_KEY");
// $config->setAccessToken("YOUR_ACCESS_TOKEN");

$grouped_signers_2_signers_1 = (new Dropbox\Sign\Model\SubSignatureRequestSigner())
    ->setName("Bob")
    ->setEmailAddress("bob@example.com");

$grouped_signers_2_signers_2 = (new Dropbox\Sign\Model\SubSignatureRequestSigner())
    ->setName("Charlie")
    ->setEmailAddress("charlie@example.com");

$grouped_signers_2_signers = [
    $grouped_signers_2_signers_1,
    $grouped_signers_2_signers_2,
];

$grouped_signers_1_signers_1 = (new Dropbox\Sign\Model\SubSignatureRequestSigner())
    ->setName("Jack")
    ->setEmailAddress("jack@example.com");

$grouped_signers_1_signers_2 = (new Dropbox\Sign\Model\SubSignatureRequestSigner())
    ->setName("Jill")
    ->setEmailAddress("jill@example.com");

$grouped_signers_1_signers = [
    $grouped_signers_1_signers_1,
    $grouped_signers_1_signers_2,
];

$signing_options = (new Dropbox\Sign\Model\SubSigningOptions())
    ->setDefaultType(Dropbox\Sign\Model\SubSigningOptions::DEFAULT_TYPE_DRAW)
    ->setDraw(true)
    ->setPhone(false)
    ->setType(true)
    ->setUpload(true);

$grouped_signers_1 = (new Dropbox\Sign\Model\SubSignatureRequestGroupedSigners())
    ->setGroup("Group #1")
    ->setOrder(0)
    ->setSigners($grouped_signers_1_signers);

$grouped_signers_2 = (new Dropbox\Sign\Model\SubSignatureRequestGroupedSigners())
    ->setGroup("Group #2")
    ->setOrder(1)
    ->setSigners($grouped_signers_2_signers);

$grouped_signers = [
    $grouped_signers_1,
    $grouped_signers_2,
];

$signature_request_create_embedded_request = (new Dropbox\Sign\Model\SignatureRequestCreateEmbeddedRequest())
    ->setClientId("b6b8e7deaf8f0b95c029dca049356d4a2cf9710a")
    ->setMessage("Please sign this NDA and then we can discuss more. Let me know if you\nhave any questions.")
    ->setSubject("The NDA we talked about")
    ->setTestMode(true)
    ->setTitle("NDA with Acme Co.")
    ->setFileUrls([
        "https://www.dropbox.com/s/ad9qnhbrjjn64tu/mutual-NDA-example.pdf?dl=1",
    ])
    ->setCcEmailAddresses([
        "lawyer1@dropboxsign.com",
        "lawyer2@dropboxsign.com",
    ])
    ->setSigningOptions($signing_options)
    ->setGroupedSigners($grouped_signers);

try {
    $response = (new Dropbox\Sign\Api\SignatureRequestApi(config: $config))->signatureRequestCreateEmbedded(
        signature_request_create_embedded_request: $signature_request_create_embedded_request,
    );

    print_r($response);
} catch (Dropbox\Sign\ApiException $e) {
    echo "Exception when calling SignatureRequestApi#signatureRequestCreateEmbedded: {$e->getMessage()}";
}

File: sandbox/java/src/main/java/com/dropbox/sign_sandbox/SignatureRequestCreateEmbeddedGroupedSignersExample.java
package com.dropbox.sign_sandbox;

import com.dropbox.sign.ApiException;
import com.dropbox.sign.Configuration;
import com.dropbox.sign.api.*;
import com.dropbox.sign.auth.*;
import com.dropbox.sign.JSON;
import com.dropbox.sign.model.*;

import java.io.File;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class SignatureRequestCreateEmbeddedGroupedSignersExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((HttpBasicAuth) config.getAuthentication("api_key")).setUsername("YOUR_API_KEY");
        // ((HttpBearerAuth) config.getAuthentication("oauth2")).setBearerToken("YOUR_ACCESS_TOKEN");

        var groupedSigners2Signers1 = new SubSignatureRequestSigner();
        groupedSigners2Signers1.name("Bob");
        groupedSigners2Signers1.emailAddress("bob@example.com");

        var groupedSigners2Signers2 = new SubSignatureRequestSigner();
        groupedSigners2Signers2.name("Charlie");
        groupedSigners2Signers2.emailAddress("charlie@example.com");

        var groupedSigners2Signers = new ArrayList<SubSignatureRequestSigner>(List.of (
            groupedSigners2Signers1,
            groupedSigners2Signers2
        ));

        var groupedSigners1Signers1 = new SubSignatureRequestSigner();
        groupedSigners1Signers1.name("Jack");
        groupedSigners1Signers1.emailAddress("jack@example.com");

        var groupedSigners1Signers2 = new SubSignatureRequestSigner();
        groupedSigners1Signers2.name("Jill");
        groupedSigners1Signers2.emailAddress("jill@example.com");

        var groupedSigners1Signers = new ArrayList<SubSignatureRequestSigner>(List.of (
            groupedSigners1Signers1,
            groupedSigners1Signers2
        ));

        var signingOptions = new SubSigningOptions();
        signingOptions.defaultType(SubSigningOptions.DefaultTypeEnum.DRAW);
        signingOptions.draw(true);
        signingOptions.phone(false);
        signingOptions.type(true);
        signingOptions.upload(true);

        var groupedSigners1 = new SubSignatureRequestGroupedSigners();
        groupedSigners1.group("Group #1");
        groupedSigners1.order(0);
        groupedSigners1.signers(groupedSigners1Signers);

        var groupedSigners2 = new SubSignatureRequestGroupedSigners();
        groupedSigners2.group("Group #2");
        groupedSigners2.order(1);
        groupedSigners2.signers(groupedSigners2Signers);

        var groupedSigners = new ArrayList<SubSignatureRequestGroupedSigners>(List.of (
            groupedSigners1,
            groupedSigners2
        ));

        var signatureRequestCreateEmbeddedRequest = new SignatureRequestCreateEmbeddedRequest();
        signatureRequestCreateEmbeddedRequest.clientId("b6b8e7deaf8f0b95c029dca049356d4a2cf9710a");
        signatureRequestCreateEmbeddedRequest.message("Please sign this NDA and then we can discuss more. Let me know if you\nhave any questions.");
        signatureRequestCreateEmbeddedRequest.subject("The NDA we talked about");
        signatureRequestCreateEmbeddedRequest.testMode(true);
        signatureRequestCreateEmbeddedRequest.title("NDA with Acme Co.");
        signatureRequestCreateEmbeddedRequest.fileUrls(List.of (
            "https://www.dropbox.com/s/ad9qnhbrjjn64tu/mutual-NDA-example.pdf?dl=1"
        ));
        signatureRequestCreateEmbeddedRequest.ccEmailAddresses(List.of (
            "lawyer1@dropboxsign.com",
            "lawyer2@dropboxsign.com"
        ));
        signatureRequestCreateEmbeddedRequest.signingOptions(signingOptions);
        signatureRequestCreateEmbeddedRequest.groupedSigners(groupedSigners);

        try
        {
            var response = new SignatureRequestApi(config).signatureRequestCreateEmbedded(
                signatureRequestCreateEmbeddedRequest
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling SignatureRequestApi#signatureRequestCreateEmbedded");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}
